compiler/luci-interpreter/src/SimpleMemoryManager.cpp

```diff
@@ -30,7 +30,9 @@ void SimpleMemoryManager::allocate_memory(luci_interpreter::Tensor &tensor)
     release_memory(tensor);
   }
   const auto element_size = getDataTypeSize(tensor.element_type());
-  const auto num_elements = tensor.shape().num_elements();
+
+  // Use large_num_elements to avoid overflow
+  const auto num_elements = tensor.shape().large_num_elements();
 
   auto *data = new uint8_t[num_elements * element_size];
```

Comment:
Q) no need to check new success/fail? (as we're now allocating large size)

Reply:
Currently, SimpleMemoryManager just throws std::bad_alloc. It would be possible to wrap `new` with try-catch to improve error handling.

Reply:
>  just throws std::bad_alloc

I think this will do :)

Reply:
As `num_elements` can be negative, `new` may throw `std::bad_array_new_length`. It could be happen when the shape has `-1`, which means undefined-yet. It seems `luci-interpreter` will not distinguish the error from huge size and undefined shape at this moment.